setup.py

```diff
@@ -28,30 +28,26 @@ def get_version():
     url='https://github.com/Shopify/pyoozie',
     packages=['pyoozie'],
     install_requires=[
-        'enum34>=0.9.23',
-        'requests>=2.12.3',
-        'six>=1.10.0',
-        'typing',
-        'untangle>=1.1.0',
-        'yattag>=1.7.2',
+        'enum34 >= 0.9.23',
+        'requests >= 2.12.3',
+        'six >= 1.10.0',
+        'typing >= 3.6.1',
+        'untangle >= 1.1.0',
+        'yattag >= 1.7.2',
     ],
     extras_require={
-        'deploy': [
-            'setuptools>=0.9'
-        ],
         'test': [
-            'autopep8',
-            'flake8',
-            'mock',
-            'mypy; python_version >= "3.3"',
+            'autopep8 == 1.3.1',
+            'mock == 2.0.0; python_version < "3.3"',
+            'mypy == 0.501; python_version >= "3.3"',
             'pycodestyle == 2.2.0',
-            'pylint',
-            'pytest-cov',
-            'pytest-randomly',
-            'pytest>=2.7',
-            'requests-mock',
-            'shopify_python==0.2.2',
-            'xmltodict',
+            'pylint == 1.6.5',
```

Comment:
To encourage reproducible builds, I set `test` requirements to exact versions. I'd do the same for `install_requires` requirements but we want to allow library users some version flexibility to avoid dependency conflicts.